charts/dapr/README.md

```diff
@@ -82,6 +82,8 @@ The Helm chart has the follow configuration options that can be supplied:
 | `global.imagePullSecret`                  | Control plane service image pull secret for docker registry             | `""`                    |
 | `global.ha.enabled`                       | Highly Availability mode enabled for control plane, except for placement service | `false`             |
 | `global.ha.replicaCount`                  | Number of replicas of control plane services in Highly Availability mode  | `3`                   |
+| `global.ha.disruption.minimumAvailable`   | Minimum amount of available instances for control plane. This can either be effective count or %. | ``             |
+| `global.ha.disruption.maximumUnavailable`   | Maximum amount of instances that are allowed to be unavailable for control plane. This can either be effective count or %. | `25%`             |
```

Comment:
Quick clarifying question. This means that 25% of the capacity can be down, and it's OK, right? So if 1 out of 5 instances were down, the instance wouldn't be resurrected?

By default, Dapr in HA does 3 replicas, so this would make any failure redeploy immediately? Is this not already the default behavior or does it just change timing?

Reply:
You are confusing 2 topics: Scheduling instances and runtime state of them.

With the HPA you define how many instances you want to have up and running, in this case 3. It's up to Kubernetes to then ensure they are there, if one gets killed it will recreated them.

With the disruption you define how many may be (un)available due to, for example node upgrades, etc. That means that it will evict all pods on a give node:
- If it evicts more than 25%, it will cancel the operation since it has too few instance running of the app.
- If it is below 25%, the operation is allowed and upgrades the node. In parallel, Kubernetes will schedule new pods on either the upgraded node or another one that is available

This guarantees that pods are not fully removed due to operations and ensures the app is always up, with a reduced set of instances.

Reply:
Thanks for the detailed explanation! I was definitely thinking of this differently.

> If it evicts more than 25%, it will cancel the operation since it has too few instance running of the app.

I'm still worried about this with the default number of instances though. If I'm understanding it correctly, the default Dapr HA deployment will have 3 instances and if we tried to upgrade them, taking down 1 instance will violate the 25% making the upgrade fail. Is this the case? Or will new instances be brought up before old ones are removed?

Reply:
If it's an upgrade then Kubernetes should manage the reschedulinh since it uses a new replica set under the hood. If you try to delete 2 instances manually, it will not allow you to do that though

You can easily pull it and try it.  

But adding @yaron2 as safety net to confirm.

Reply:
You can learn more on https://kubernetes.io/docs/concepts/workloads/pods/disruptions/ btw

Reply:
I will pull this PR and try this out. If works as expected we can merge.

Reply:
This works well. We'll be sure to test this in our e2e upgrade/install scenarios.